File: app/back/src/main/java/mitei/mitei/investigate/report/balance/politician/dto/SelectOptionDto.java
package mitei.mitei.investigate.report.balance.politician.dto;

import java.io.Serializable;

/**
 * HTMLにおけるSelectボタンのOption項目を表すDto.
 */
public class SelectOptionDto implements Serializable { // NOPMD DataClass

    /** Serialize id. */
    private static final long serialVersionUID = 1L;

    /** option項目のvalue(値). */
    private String value;

    /** option項目の表示テキスト. */
    private String text;

    /**
     * オプション項目の値を取得する.
     *
     * @return オプション項目の値
     */
    public String getValue() {
        return value;
    }

    /**
     * オプション項目の値を設定する.
     *
     * @param value オプション項目の値
     */
    public void setValue(final String value) {
        this.value = value;
    }

    /**
     * オプション項目のテキストを取得する.
     *
     * @return オプション項目のテキスト
     */
    public String getText() {
        return text;
    }

    /**
     * オプション項目のテキストを設定する.
     *
     * @param text オプション項目のテキスト
     */
    public void setText(final String text) {
        this.text = text;
    }

}

File: app/front/src/dto/selectOptionDto.ts
export default interface SelectOptionInterface {
    value: string;
    text: string;
}

/**
 * HTMLにおける、Selectボタンの選択(option)項目データDto
 *
 * @class SelectOptionDto
 */
export default class SelectOptionDto implements SelectOptionInterface{

    /**
     * option項目の値
     * @type {string}
     * @memberof SelectOption
     */
    value: string;

    /**
     * option項目の表示内容
     * @type {string}
     * @memberof SelectOption
     */
    text: string;

    /**
     * Creates an instance of SelectOption.
     * @param {string} value option項目に設定したい値
     * @param {string} text option項目に設定したいテキスト
     * @memberof SelectOption
     */
    constructor() {
        this.value = "";
        this.text = "";
    }
}
